google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -117,7 +117,8 @@ protected void doStop() {
 
     lock.lock();
     try {
-      requestObserver.onError(Status.CANCELLED.asException());
+      // Don't onError; it appears erroring can cause gRPC to drop messages.
+      requestObserver.onCompleted();
     } finally {
```

Comment:
I wouldn't use on completed, if the server hangs you may never actually complete the RPC. Maybe schedule the onerror so it runs after acks?

Reply:
I think we're already running `onError` after acks. The problem was previously that `onError` can caused buffered messages to get dropped. Now we're using unary calls to ack, so there's no problem anymore. I'll revert this.